lib/resty/core/regex.lua

```diff
@@ -479,6 +480,71 @@ function ngx.re.find(subj, regex, opts, ctx, nth)
 end
 
 
+local function iterate_re_gmatch(self)
+    local compile_once = self._compile_once
```

Comment:
Is it worth it to dereference this variable on each iteration, considering we will only use it, once, on the last iteration?